### PR TITLE
Adding comments to Protobuf

### PR DESCRIPTION
**Pull request by KidiH**
Previously, I had java classes add the comments to datastore. Now, the use of protobufs gives the added benefit of smaller storage requirement that come in very handy. 
The changes made were:
- Update the DataServlet.java file to accept and pass protobufs 
- Update the script.js file to print out protobufs

### CODE REVIEW

**Comment by KidiH** [comment]
I accidentally added a chart I had been experimenting with in commit 5cdbee6.  I will update this soon.

File: portfolio/src/main/java/com/google/sps/data/TotalCommentProtos.java [hunk 1]
```diff
@@ -0,0 +1,1783 @@
+// Generated by the protocol buffer compiler.  DO NOT EDIT!
```
Comment (ArunChauhan) [inline portfolio/src/main/java/com/google/sps/data/TotalCommentProtos.java:1]: This file is auto-generated from the proto file. Please do not add it to the repo. In general, we only want to add the minimal sources from which your portfolio site may be reconstructed.

File: portfolio/src/main/java/com/google/sps/data/comments.proto [hunk 2]
```diff
@@ -0,0 +1,16 @@
+syntax = "proto2";
+
+package tutorial;
+
+option java_package = "com.google.sps.data";
+option java_outer_classname = "TotalCommentProtos";
+
+message Comment {
+  optional string name = 1;
+  required string data = 2;
```
Comment (ArunChauhan) [inline portfolio/src/main/java/com/google/sps/data/comments.proto:10]: The Google style guide discourages using 'required' fields. It seems counterintuitive but it has to do with long term maintainability of code. This is only FYI, you don't have to change anything here for this pull request.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 3]
```diff
@@ -1,4 +1,3 @@
-// Copyright 2019 Google LLC
```
Comment (ArunChauhan) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:1]: Please make sure you run the formatter on the file.

File: portfolio/src/main/webapp/script.js [hunk 4]
```diff
@@ -65,22 +67,36 @@ async function getRandomProverb(){
...
+        }
 });
 }
 
 /** Creates an <p> element containing text. */
 function createListElement(text) {
+  const liElem = document.createElement('li');
+
+  const heading = document.createElement('h5');
+  if(String.fromCharCode.apply(String, text.name_.bytes) === ""){
```
Comment (ArunChauhan) [inline portfolio/src/main/webapp/script.js]: I am not sure if there a Google style formatter for JavaScript, but I would suggest leaving a blank space between 'if' and '('.  Similarly, leave a blank space between ')' and '{'.

File: portfolio/src/main/webapp/script.js [hunk 5]
```diff
@@ -65,22 +67,36 @@ async function getRandomProverb(){
...
+      heading.innerText = "Anonymous";
+  }
+  else{
```
Comment (ArunChauhan) [inline portfolio/src/main/webapp/script.js]: Add space between 'else' and '{'.

**Review by ArunChauhan — COMMENTED** [review]
Why is the file "protobuf-java-3.12.2 (1).tar.gz" part of the pull request?

File: portfolio/src/main/java/com/google/sps/data/TotalCommentProtos.java (hunk 1 above)
Comment (KidiH) [reply portfolio/src/main/java/com/google/sps/data/TotalCommentProtos.java:1]: Thank you! Removed the file!

File: portfolio/src/main/java/com/google/sps/data/comments.proto (hunk 2 above)
Comment (KidiH) [reply portfolio/src/main/java/com/google/sps/data/comments.proto:10]: Thank you! I will keep that in mind.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 3 above)
Comment (KidiH) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:1]: Updated!

File: portfolio/src/main/webapp/script.js (hunk 4 above)
Comment (KidiH) [reply portfolio/src/main/webapp/script.js]: Corrected!

File: portfolio/src/main/webapp/script.js (hunk 5 above)
Comment (KidiH) [reply portfolio/src/main/webapp/script.js]: Corrected!

**Review by ArunChauhan — APPROVED** [review]

**Review by cshei — APPROVED** [review]

**Pull request merged**